examples/systemcalls/systemcalls.c

```diff
@@ -34,66 +53,162 @@ bool do_system(const char *cmd)
- *   (first argument to execv), and use the remaining arguments
- *   as second argument to the execv() command.
- *
-*/
 
-    va_end(args);
+    char *path = command[0];
+
+    //Ensure that the path is not null or empty
+    if (path == NULL || strlen(path) == 0) {
+        syslog(LOG_ERR, "AESD: Path is empty or NULL\n");
+        return false;
+    }
+
+    //Check if the path is valid - MOdified based on CHATGPT output
+    struct stat buffer;
+    if (stat(path, &buffer) != 0) {
+        syslog(LOG_ERR, "AESD: Path is invalid\n");
+        return false;
+    }
+
+    //Create new process
+    pid_t child_pid = fork();
+
+    if (child_pid == -1) {
+        syslog(LOG_ERR, "AESD: Failed to fork a child process\n");
+        exit(EXIT_FAILURE);
+        return false;
+    } else if (child_pid == 0) {
+
+        // Execute the command and check if the process returns okay
+        if (execv(path, command) == -1) {
+            syslog(LOG_ERR, "AESD: Program failed to execute\n");
+            exit(EXIT_FAILURE);
+        }
+        // syslog(LOG_INFO, "AESD: Child Process Completed\n");
+        //exit(EXIT_SUCCESS);
+    } else {
+        // Wait for the child process to complete
+        int status;
+        // if (waitpid(child_pid, &status, 0) == -1) {
+        if (wait(&status) == -1) {
+            syslog(LOG_ERR, "AESD: Wait failed!\n");
+            return false;
+        }
+
+        if (WIFEXITED(status)) {
+            int exit_status = WEXITSTATUS(status);
+            if (exit_status != 0) {
+                return false;
+            }
+        } else {
+            return false; // Command execution failed
+        }
+    }
 
+    va_end(args);
     return true;
 }
-
 /**
 * @param outputfile - The full path to the file to write with command output.
 *   This file will be closed at completion of the function call.
 * All other parameters, see do_exec above
 */
-bool do_exec_redirect(const char *outputfile, int count, ...)
-{
+//Reference used: https://stackoverflow.com/questions/13784269/redirection-inside-call-to-execvp-not-working/13784315#13784315
+bool do_exec_redirect(const char *outputfile, int count, ...) {
+    openlog(NULL, 0, LOG_USER);
+    syslog(LOG_INFO, "AESD: BEGIN LOG");
+    int fd;
     va_list args;
     va_start(args, count);
-    char * command[count+1];
+    char *command[count + 1];
     int i;
-    for(i=0; i<count; i++)
-    {
-        command[i] = va_arg(args, char *);
+    for (i = 0; i < count; i++) {
+        command[i] = va_arg(args,
+        char *);
     }
     command[count] = NULL;
-    // this line is to avoid a compile warning before your implementation is complete
-    // and may be removed
-    command[count] = command[count];
 
 
-/*
- * TODO
- *   Call execv, but first using https://stackoverflow.com/a/13784315/1446624 as a refernce,
- *   redirect standard out to a file specified by outputfile.
- *   The rest of the behaviour is same as do_exec()
- *
-*/
+    char *path = command[0];
```

Comment:
@Silex93 think about how you could avoid duplicating logic between these two functions.